src/components/text-editor/prosemirror-adapter/plugins/link-plugin.ts

```diff
@@ -0,0 +1,214 @@
+import { Plugin, PluginKey, TextSelection } from 'prosemirror-state';
+import { EditorView } from 'prosemirror-view';
+import { schema } from 'prosemirror-schema-basic';
+import { Mark } from 'prosemirror-model';
+import { isExternalLink, isValidUrl } from '../menu/menu-commands';
+import { EditorMenuTypes } from '../menu/types';
+
+export const linkPluginKey = new PluginKey('linkPlugin');
+
+export type UpdateLinkCallback = (text: string, href: string) => void;
+
+export interface EditorLinkMenuEventDetail {
+    href: string;
+    text: string;
+}
+
+const updateLink = (
+    view: EditorView,
+    updateLinkCallback?: UpdateLinkCallback,
+) => {
+    const { from, to } = view.state.selection;
+
+    let text = '';
+    let href = '';
+    view.state.doc.nodesBetween(from, to, (node) => {
+        if (node.type.name === 'text') {
+            text = node.text;
+            node.marks.forEach((mark: Mark) => {
+                if (mark.type.name === 'link') {
+                    href = mark.attrs.href;
+                }
+            });
+        }
+    });
+
+    if (updateLinkCallback) {
+        updateLinkCallback(text, href);
+    }
+};
+
+export const createLinkPlugin = (updateLinkCallback?: UpdateLinkCallback) => {
+    return new Plugin({
+        key: linkPluginKey,
+        props: {
+            handlePaste: (view, event) => {
+                return processPasteEvent(view, event);
+            },
+            handleDOMEvents: {
+                mousedown: (view, event) => {
+                    if (
+                        (event.metaKey || event.ctrlKey) &&
+                        event.button === 0
+                    ) {
+                        return processModClickEvent(view, event);
+                    }
+
+                    return processClickEvent(view, event);
```

Comment:
Should we call this `processDoubleClickEvent` in line with the `processModClickEvent` function?

Reply:
⚡

Reply:
⚡